src/main/java/htsjdk/samtools/SAMRecordUtil.java

```diff
@@ -64,8 +79,13 @@ public static void reverseComplement(final SAMRecord rec, final Collection<Strin
             for (final String tag: tagsToRevcomp) {
                 Object value = rec.getAttribute(tag);
                 if (value != null) {
-                    if (value instanceof byte[]) SequenceUtil.reverseComplement((byte[]) value);
-                    else if (value instanceof String) value = SequenceUtil.reverseComplement((String) value);
+                    if (value instanceof byte[]) {
+                        value = unsafe ? value : ((byte[]) value).clone();
+                        SequenceUtil.reverseComplement((byte[]) value);
+                    }
+                    else if (value instanceof String) {
+                        value = SequenceUtil.reverseComplement((String) value);
```

Comment:
perhaps a comment here for the reader that `SequenceUtil.reverseComplement` is in-place with bytes and copies with `String`.